main.rb

```diff
@@ -0,0 +1,118 @@
+module Enumerable
+  
+  # my each
+  def my_each
+    count = 0
+    while count < length
+      yield(self[count])
+      count += 1
+    end
+    self
+  end
+  
+  # my_each_with_each
+  def my_each_with_index
+    count = 0
+    while count < length
+      yield(self[count], count)
+      count += 1
+    end
+    self
+  end
+  
+  # my_select
+  def my_select
+    count = 0
+    arr = []
+    while count < length
+      arr.push(self[count]) if yield(self[count])
+      count += 1
+    end
+    arr
+  end
+  
+  # my_all
+  def my_all?
```

Comment:
## Your `my_all?` method apparently is not able to return a proper result.

This was tested trying on an array of 10 number 1 integers, running the following command `array.my_all? { |n| n.is_a? Integer } ` returning false, whilst it should return true. Passing the following block `{ |n| n < 2 }` returns an error. I will ask of you here, to take a closer look at your code, you will spot an inconsistency, you are starting your while loop looking at index 1, not 0. Try to check what are you exactly yielding.